### ISSUE TITLE
A problem about position ids of blocks which are not the anchor block?

### ISSUE THREAD

**Issue by YongqiQiao** [NVIDIA/Star-Attention#4]
As mentioned in the paper, the position ids of anchor block is set to be the same as first block. Then how those blocks'position ids which are not the anchor block, change? For example, the host 1 hosts c1, and c3,  whether is the c3's position ids set to be its original position ids?

**Comment by shan18** [comment]
Yes, the position IDs for each block remains same as in the original prompt.

We create the position IDs based on the context length
https://github.com/NVIDIA/Star-Attention/blob/443bb9340555af7da192df7f733001f5f00382d7/model.py#L193

Then, when splitting the context block, we also split the corresponding position IDs
https://github.com/NVIDIA/Star-Attention/blob/443bb9340555af7da192df7f733001f5f00382d7/model.py#L246-L250

This way, for example if the block size is 4K, then for c1 the position ID is (0, 4K) and for c3 it would be (8K, 12K).
